tests/meshcop/commissioner.hpp

```diff
@@ -0,0 +1,309 @@
+#include "web/pskc-generator/pskc.hpp"
+
+
+using namespace ot;
+using namespace ot::Utils;
+using namespace ot::BorderRouter;
+
+
+#define MBEDTLS_DEBUG_C
+#define DEBUG_LEVEL 1
+
+#define SYSLOG_IDENT "otbr-commissioner"
+
+#include "commissioner_argcargv.hpp"
+
+/**
+ * Constants
+ */
+enum
+{
+    /* max size of a network packet */
+    kSizeMaxPacket = 1500,
+
+    /* Default size of steering data */
+    kSteeringDefaultLength = 15,
+
+    /* how long is an EUI64 in bytes */
+    kEui64Len = (64 / 8),
+
+    /* how long is a PSKd in bytes */
+    kPSKdLength = 32,
+
+    /* What port does our internal server use? */
+    kPortJoinerSession = 49192,
+
+    /* 64bit xpanid length in bytes */
+    kXpanidLength = (64 / 8), /* 64bits */
+
+    /* specification: 8.10.4 */
+    kNetworkNameLenMax = 16,
+
+    /* Spec is not specific about this items max length, so we choose 64 */
+    kBorderRouterPassPhraseLen = 64,
+
+};
+
+
+
+/**
+ * Commissioner State
+ */
+enum
+{
+    kStateInvalid,
+    kStateConnected,
+    kStateAccepted,
+    kStateReady,
+    kStateAuthenticated,
+    kStateFinalized,
+    kStateDone,
+    kStateError,
+};
+
+
+/**
+ * Commissioner TLV types
+ */
+enum
+{
+    kJoinerDtlsEncapsulation
+};
+
+
+/**
+ * Commissioner Context.
+ */
+struct Context
+{
+    /** Set to true if we should commission the device */
+    bool commission_device;
+
+    /** coap instance to talk to agent & device */
+    Coap::Agent *mCoap;
+
+    /** dtls info for talking to agent & joiner */
+    Dtls::Server *mDtlsServer;
+
+    /* the session info for agent & joiner */
+    Dtls::Session *mSession;
+
+    /* dtls context information */
+    mbedtls_ssl_context *mSsl;
+    mbedtls_net_context *mNet;
+
+    /* Socket we are using with the agent */
+    int mSocket;
+
+    /* this generates our coap tokens */
+    uint16_t mCoapToken;
+
+    /* this is the commissioner session id */
+    uint16_t mCommissionerSessionId;
+
+    /** all things for the border agent */
+    struct br_agent
+    {
+
+        /** network port, in form mbed likes it (ascii) */
+        char mPort_ascii[ 7 ];
+
+        /** network address, in form mbed likes it (ascii) */
+        char mAddress_ascii[64];
+
+        /** xpanid from command line & binary form */
+        /* Note: xpanid is used to calculate the PSKc */
+        struct
+        {
+            char    ascii[  (kXpanidLength * 2) + 1 ];
+            uint8_t bin  [   kXpanidLength ];
+        } mXpanid;
+
+        /** network name from command line */
+        /* Note: networkname is used to calculate the PSKc */
+        char mNetworkName[ kNetworkNameLenMax + 1 ];
+
+        /** border router agent pass phrase */
+        /* Note: passphrase is used to calculate the PSKc */
+        char mPassPhrase[ kBorderRouterPassPhraseLen + 1 ];
+
+
+        /** the PSKc used with the agent */
+        struct
+        {
+
+            /* this class does the calculation */
+            Psk::Pskc mTool;
```

Comment:
As OpenThread style guide suggested
>Data objects shall be declared at the smallest possible level of scope.

This `mTool` was used only once at a function. I think it should be the function(i.e.`compute_pskc()`)'s local scope variable.